### PR TITLE
allpass module

### PR DESCRIPTION
**Pull request by beserge**
Brought up to speed with master.

### CODE REVIEW

**Comment by stephenhensley** [comment]
Update the documentation so that it matches doxygen formatting (interestingly this won't register as being documented, and therefore won't throw errors).

check through some of the other files for syntax as necessary, but pretty much 

```C++
/** Description

    Any other details
     
    \param foo array
    \param size number of elements in foo or whatever
    \return a magic spell designed by foo runes
*/
magic_spell create_spell(void* foo, size_t size);
```

**Comment by TheSlowGrowth** [comment]
We should eventually enable `WARN_AS_ERROR` in the doxyfile. Then the check will fail on undocumented code. I had left this off because it would have made the transition to doxygen quite painful.
ATM there are a few other warnings that will have to be resolved before we can turn `WARN_AS_ERROR` on.

**Comment by stephenhensley** [comment]
@TheSlowGrowth yeah, I figured there was a setting for that in the Doxyfile.

On my dev machines I don't get any warnings from rebuilding docs, but I do see warnings on the CI. Mostly for invalid keys (which I'm guessing is a version discrepancy).

File: modules/allpass.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/*  Allpass
+    Allpass filter module
+  
+    Passes all frequencies at their original levels, with a phase shift.
+  
+    Original author(s) : Barry Vercoe, John ffitch
+    Year : 1991
+    Ported from soundpipe by Ben Sergentanis, May 2020
+*/
```
Comment (stephenhensley) [inline modules/allpass.h:9]: should be `/** Allpass filter module`, and should move down to line22 where current start of class is. No need to have the class name as Doxygen will grab that from the class definition.

**Comment by stephenhensley** [comment]
Might sound a bit better with a slower LFO, but no more pop. So that's good.

**Pull request merged**